## Instructions:
### get spectra data of one galaxy 
* Use center position $skycord$
* creating subcube with size c $\pm 50$ pixel (100 $\times$ 100 )
  >Creating a New FITS File with name *coord_small.fits*


In [1]:
import numpy as np
import matplotlib.pyplot as plt
import astropy.units as u
from mpdaf.obj import Cube, WCS, WaveCoord
#from astropy import units as u
from astropy.coordinates import SkyCoord
#%matplotlib notebook 
%matplotlib inline 

### 4C+19.71

In [8]:
c = SkyCoord('21h44m07.56s', '+19d29m14.6s', frame='icrs')
print('degree ={}'.format(c))

degree =<SkyCoord (ICRS): (ra, dec) in deg
    (326.0315, 19.48738889)>


In [9]:
RG3 = Cube('/Users/Ray/Desktop/Uni/8.Bachelorarbeit/Data/Cubes.2023/DATACUBE_4C19_ZAPmask_pipesky_fullFOV_mad_AstroGaia_varscaled_corr_zero.fits')
#RG3.info()

In [10]:
ra, dec = (326.0315, 19.48738889)
pix = RG3.wcs.sky2pix([dec,ra], unit=u.deg) 
x = int(np.around(pix[0][1]))
y = int(np.around(pix[0][0]))
print(y,x)
#make a copy of the (sub-)cube, prevent changing the original cube
cube_new_3 = RG3.copy()

167 147


In [ ]:
cube_new_3[:,167-50:167+50,147-50:147+50].write('4C+19.71_small.fits', savemask='dq')

### MRC 0316-257

In [3]:
c = SkyCoord('3h18m12.07s', '-25d35m10.22s', frame='icrs')
print('degree ={}'.format(c))

degree =<SkyCoord (ICRS): (ra, dec) in deg
    (49.55029167, -25.58617222)>


In [4]:
RG4 = Cube('/Users/Ray/Desktop/Uni/8.Bachelorarbeit/Data/Cubes.2023/DATACUBE_MRC0316_ZAPmask_pipesky_sigma6_mad_AstroHSTGaia_varscaled_corr_zero.fits')
#RG4.info()

In [5]:
ra, dec = (49.55029167, -25.58617222)
pix = RG4.wcs.sky2pix([dec,ra], unit=u.deg) 
x = int(np.around(pix[0][1]))
y = int(np.around(pix[0][0]))
print(y,x)
#make a copy of the (sub-)cube, prevent changing the original cube


189 145


In [7]:
cube_new_4 = RG4.copy()
cube_new_4[:,189-50:189+50,145-50:145+50].write('MRC0316_small.fits', savemask='dq')

### MRC 0943-242

In [2]:
c = SkyCoord('9h45m32.73s', '-24d28m49.65s', frame='icrs')
print('degree ={}'.format(c))

degree =<SkyCoord (ICRS): (ra, dec) in deg
    (146.386375, -24.48045833)>


In [3]:
RG5 = Cube('/Users/Ray/Desktop/Uni/8.Bachelorarbeit/Data/Cubes.2023/DATACUBE_MRC0943_ZAPmask_pipesky_sigma6_mad_v1_AstroGaia_varscaled_corr_zero.fits')
#RG5.info()

In [4]:
ra, dec = (146.386375, -24.48045833)
pix = RG5.wcs.sky2pix([dec,ra], unit=u.deg) 
x = int(np.around(pix[0][1]))
y = int(np.around(pix[0][0]))
print(y,x)
#make a copy of the (sub-)cube, prevent changing the original cube

212 158


In [5]:
cube_new_5 = RG5.copy()
cube_new_5[:,212-50:212+50,158-50:158+50].write('MRC0943_small.fits', savemask='dq')

### TN J0121+1320

In [2]:
c = SkyCoord('1h21m42.73s', '13d20m58.0s', frame='icrs')
print('degree ={}'.format(c))

degree =<SkyCoord (ICRS): (ra, dec) in deg
    (20.42804167, 13.34944444)>


In [3]:
RG6 = Cube('/Users/Ray/Desktop/Uni/8.Bachelorarbeit/Data/Cubes.2023/DATACUBE_TNJ0121_ZAPmask_pipesky_fullFOV_mad_10EXP_AstroSDSS_varscaled_corr_zero.fits')
#RG6.info()

In [4]:
ra, dec = (20.42804167, 13.34944444)
pix = RG6.wcs.sky2pix([dec,ra], unit=u.deg) 
x = int(np.around(pix[0][1]))
y = int(np.around(pix[0][0]))
print(y,x)
#make a copy of the (sub-)cube, prevent changing the original cube

149 109


In [5]:
cube_new_6 = RG6.copy()
cube_new_6[:,149-50:149+50,109-50:109+50].write('TNJ0121_small.fits', savemask='dq')

### TN J0205+2242

In [2]:
c = SkyCoord('2h05m10.69s', '22d42m50.4s', frame='icrs')
print('degree ={}'.format(c))

degree =<SkyCoord (ICRS): (ra, dec) in deg
    (31.29454167, 22.714)>


In [3]:
RG7 = Cube('/Users/Ray/Desktop/Uni/8.Bachelorarbeit/Data/Cubes.2023/DATACUBE_TNJ0205_ZAPmask_pipesky_fullFOV_mad_Astro_scaled_corr_zero.fits')
#RG7.info()

In [4]:
ra, dec = (31.29454167, 22.714)
pix = RG7.wcs.sky2pix([dec,ra], unit=u.deg) 
x = int(np.around(pix[0][1]))
y = int(np.around(pix[0][0]))
print(y,x)
#make a copy of the (sub-)cube, prevent changing the original cube

174 174


In [5]:
cube_new_7 = RG7.copy()
cube_new_7[:,174-50:174+50,174-50:174+50].write('TNJ0205_small.fits', savemask='dq')

### TN J1338-1942

In [2]:
c = SkyCoord('13h38m26.10s', '-19d42m31.1s', frame='icrs')
print('degree ={}'.format(c))

degree =<SkyCoord (ICRS): (ra, dec) in deg
    (204.60875, -19.70863889)>


In [3]:
# only read the center part of the cube
x = 331
y = 233
RG8 = Cube('/Users/Ray/Desktop/Uni/8.Bachelorarbeit/Data/Cubes.2023/DATACUBE_TNJ1338_ZAPmask_pipesky_mad_AstroGaia_varscaled_corr_zero.fits')[:,y-50:y+50,x-50:x+50]
#RG8.info()

ra, dec = (204.60875, -19.70863889)
pix = RG8.wcs.sky2pix([dec,ra], unit=u.deg) 
x = int(np.around(pix[0][1]))
y = int(np.around(pix[0][0]))
print(y,x)


In [4]:
#make a copy of the (sub-)cube, prevent changing the original cube
#cube_new_8 = RG8.copy()
RG8.write('TNJ1338_small.fits', savemask='dq')